waiter/src/waiter/auth/authentication.clj

```diff
@@ -379,16 +380,37 @@
       :else
       (on-auth-required))))
 
-(defn wrap-auth-bypass
-  "Middleware that checks if authentication is disabled for a token and sets the
-  :skip-authentication key of the request to true before passing to next handler."
-  [handler]
+(defn- make-wrap-auth-bypass
+  "Takes a handler and a function to call when there is an error with processing the
+  authentication parameter. If successful and authentication is disabled for the token,
+  then the :skip-authentication is set to true for the request object before getting passed
+  to the handler"
+  [handler on-error]
   (fn auth-bypass-handler [{:keys [waiter-discovery] :as request}]
     (process-authentication-parameter
       waiter-discovery
-      (fn [status message]
-        (utils/clj->json-response {:error message} :status status))
-      (fn []
+      (fn on-process-authentication-error [status message]
+        (on-error request status message))
+      (fn on-auth-disabled []
         (handler (assoc request :skip-authentication true)))
-      (fn []
-        (handler request)))))
\ No newline at end of file
+      (fn on-auth-enabled []
+        (handler request)))))
+
+(defn wrap-auth-bypass
+  "Middleware that checks if authentication is disabled for a token and sets the
+  :skip-authentication key of the request to true before passing to next handler."
+  [handler]
+  (make-wrap-auth-bypass
+    handler
+    (fn on-error [_ status message]
```

Comment:
Each of these functions need unique names if they are in the same file :(
E.g. `on-http-error` and `on-ws-error`